src/Orchard.Cms.Web/Modules/Orchard.Contents/Controllers/AdminController.cs

```diff
@@ -474,6 +475,39 @@ public Task<IActionResult> EditPOST(string contentItemId, string returnUrl)
                 return Unauthorized();
             }
 
+            ContentTypeDefinition typeDefinition = _contentDefinitionManager.GetTypeDefinition(contentItem.ContentType);
```

Comment:
I don't think the logic should be in the controller. What if we use the API directly?